ui/page/dexclient/markets.go

```diff
@@ -0,0 +1,186 @@
+package dexclient
+
+import (
+	"context"
+	"fmt"
+
+	"decred.org/dcrdex/client/core"
+	"gioui.org/layout"
+
+	"github.com/planetdecred/godcr/ui/decredmaterial"
+	"github.com/planetdecred/godcr/ui/load"
+	"github.com/planetdecred/godcr/ui/modal"
+	"github.com/planetdecred/godcr/ui/page/components"
+	"github.com/planetdecred/godcr/ui/values"
+)
+
+type (
+	C = layout.Context
+	D = layout.Dimensions
+)
+
+const MarketPageID = "Markets"
+
+type Page struct {
+	*load.Load
+	ctx       context.Context // page context
+	ctxCancel context.CancelFunc
+
+	addDexWidget *addDexWidget
+	login        decredmaterial.Button
+	initialize   decredmaterial.Button
+}
+
+// TODO: Add collapsible button to select a market.
+// Use mktName="DCR-BTC" in the meantime.
+func (pg *Page) dexMarket(mktName string) *core.Market {
+	dex := pg.dex()
+	if dex == nil {
+		return nil
+	}
+	return dex.Markets[mktName]
+}
+
+func NewMarketPage(l *load.Load) *Page {
+	pg := &Page{
+		Load:         l,
+		addDexWidget: newAddDexWidget(l),
```

Comment:
Should only create this widget when necessary. Don't want to load this widget into memory when a DEX has already been added and it won't ever be used.

Reply:
Changed it for now, but in the future can add more dex. So we'll need to init in page struct